lib/child_process.js

```diff
@@ -484,10 +474,12 @@ function spawnSync(/*file, args, options*/) {
   // Validate the kill signal, if present.
   if (typeof options.killSignal === 'string' ||
       typeof options.killSignal === 'number') {
-    options.killSignal = lookupSignal(options.killSignal);
 
     if (options.killSignal === 0)
       throw new RangeError('"killSignal" cannot be 0');
+
+    options.killSignal = convertToValidSignal(options.killSignal);
```

Comment:
When you rebase this, can you move the `convertToValidSignal()` calls into `validateKillSignal()`.